src/rich-text-parser.js

```diff
@@ -0,0 +1,25 @@
+import { NodeTree } from './node-tree'
+
+export class RichTextParser {
+  /**
+   * Parse Prismic rich text components into HTML string.
+   *
+   * @param richText - Prismic rich text component
+   * @param htmlSerializer - HtmlSerializer used for parsing
+   * @returns {string} Rich text as HTML string
+   */
+  static parseAsHtml(richText, htmlSerializer) {
+    return htmlSerializer.serializeNodeTree(new NodeTree(richText)).join('')
+  }
```

Comment:
I wondered why creating an object that only returns an arr from a constructor and then join. Why not just a method and also put join there?

Reply:
This is for separation of concerns. I think we can argue about the reason for a separate NodeTree class, but mainly this is used as an interface, so that the serializer can know which object to expect.
This way we can also give a user the possibility to provide its own serializer.
A pretty weak reason for this approach, but still valid, is to keep the interface closer to prismic-dom, where you are also exposed a parser with these two methods.

Also the join is for the serialization result not the node tree children

Reply:
And why use getChildren in the stance?